### PR TITLE
Add BMP388 / BMP3xx

### PR DESCRIPTION
**Pull request by tyeth**
Adds BMP388, tested with two connected to QTPYs2.
image
image

### CODE REVIEW

**Comment by tyeth** [comment]
@lorennorman Ready for review, then #462 please :) I think Brents going to be away most of this week, so look out for some component definitions in the next few days (probably Wednesday)

**Review by brentru — CHANGES_REQUESTED** [review]

**Comment by tyeth** [comment]
@brentru should be good now. Retested too.
image

File: src/components/i2c/WipperSnapper_I2C.h [hunk 1]
```diff
@@ -100,6 +101,7 @@ class WipperSnapper_Component_I2C {
   WipperSnapper_I2C_Driver_BH1750 *_bh1750 = nullptr;
   WipperSnapper_I2C_Driver_BME280 *_bme280 = nullptr;
   WipperSnapper_I2C_Driver_BMP280 *_bmp280 = nullptr;
+  WipperSnapper_I2C_Driver_BMP3XX *_bmp388 = nullptr;
```
Comment (brentru) [inline src/components/i2c/WipperSnapper_I2C.h:104]: here too and don;'t forget to change _i2c.cpp as well

Comment (tyeth) [reply src/components/i2c/WipperSnapper_I2C.h:104]: :) Read my slightly slow mind, just committing and retesting now

**Comment by tyeth** [comment]
@brentru retested and resolved.
Just a thought, seems to not send a value initially until the polling period is expired (leaving at 15min default). Setting to 30seconds seems to trigger a send but not quite sure as not looked at that bit of code in context. Would be nice if it always sent the first values irrespective of the poll periods.
image

**Comment by brentru** [comment]
> Would be nice if it always sent the first values irrespective of the poll periods.

Agreed, want to open an issue for this functionality?

**Review by brentru — APPROVED** [review]

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_BMP3XX.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
+/*!
+ * @file WipperSnapper_I2C_Driver_BMP3XX.h
+ *
+ * Device driver for an AHT Humidity and Temperature sensor.
```
[the remark this answers is not in the document]
Comment (tyeth) [reply src/components/i2c/drivers/WipperSnapper_I2C_Driver_BMP3XX.h]: I'll make sure I'm less copy paste happy in future. Too many of these in this PR, sorry!

[the remark this answers is not in the document]
Comment (brentru) [reply src/components/i2c/drivers/WipperSnapper_I2C_Driver_BMP3XX.h]: It's ok and why we have code reviews!

**Comment by tyeth** [comment]
> > Would be nice if it always sent the first values irrespective of the poll periods.
> 
> Agreed, want to open an issue for this functionality?

https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/issues/463

**Pull request merged**